ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,383 @@
+      }
+      return this.state.allExamples[currentIndex];
+    }else{
+      return null;
+    }
+  },
+
+  logResponse() {
+    this.logData('message_popup_response');
+  },
+  
+  logRevision(finalText) {
+    this.logData('message_popup_revision', {finalResponseText: finalText});
+  },
+  
+  logRevisionDeclined(){
+    this.logData('message_popup_revision_declined');
+  },
+  
+  logData(type, params = {}) {
+    const {elapsedMs, initialResponse} = this.state;
+    const {question, shouldShowStudentCard, helpType} = this.props;
+    const {finalResponseText} = params;
+    const response:Response = {
+      question,
+      shouldShowStudentCard,
+      helpType,
+      elapsedMs,
+      initialResponse,
+      finalResponseText
+    };
+    this.props.onLog(type, response);
+  },
+
+  resetExperience(){
+    this.setState(this.getInitialState());
+    this.props.resetExperience();
+  },
+
+  getInitialMessages(nextProps){
+    var messages = [];
+    if(nextProps.hasStarted && nextProps.questionsAnswered < nextProps.sessionLength){
+      const question = {
+        type: 'info',
+        text: nextProps.question.text,
+        key: 'question'
+      };
+      messages.push(question);
+      if(nextProps.shouldShowStudentCard && _.has(nextProps.question, 'studentId')){
+        const studentObject = (withStudents([nextProps.question]))[0].student
+        var text = studentObject.name + "\n"
+                      + studentObject.grade + " " + studentObject.gender + ", " + studentObject.race;
+        if(_.has(studentObject, 'behavior') && studentObject.behavior !== null){ text+= "\n" + studentObject.behavior };
+        if(_.has(studentObject, 'ell') && studentObject.ell !== null){ text+= "\n" + studentObject.ell };
+        if(_.has(studentObject, 'learningDisabilities') && studentObject.learningDisabilities !== null){ text+= "\n" + studentObject.learningDisabilities };
+        if(_.has(studentObject, 'academicPerformance') && studentObject.academicPerformance !== null){ text+= "\n" + studentObject.academicPerformance };
+        if(_.has(studentObject, 'interests') && studentObject.interests !== null){ text+= "\n" + studentObject.interests };
+        if(_.has(studentObject, 'familyBackground') && studentObject.familyBackground !== null){ text+= "\n" + studentObject.familyBackground };
+        if(_.has(studentObject, 'ses') && studentObject.ses !== null){ text+= "\n" + studentObject.ses };
+        const student = {
+          type: 'student',
+          text: text,
+          key: 'studentInfo'
+        }
+        messages.push(student);
+      }
+    }
+    return messages;
+  },
+  
+  componentDidMount() {
+    setInterval(this.updateTimer, ONE_SECOND);
+  },
+
+  updateTimer() {
+    if(!this.isReadyToMoveOn()){
+      this.setState({ elapsedMs: this.state.elapsedMs + ONE_SECOND });
+    }
+  },
+  
+  componentWillMount(){
+    this.messageId = parseInt(this.props.question.id)*10 + 1;
```

Comment:
I'm not quite sure what this is about - do messages need an id in order to use it as a key when rendering?  I was thinking that the message index might be sufficient for that purpose.  Or, if you want to make a unique message id, you can used the `uuid` package.  I'm not sure why it's related to the `question.id` here though.